ext/RMagick/rminfo.c

```diff
@@ -2638,8 +2638,15 @@ Info_initialize(VALUE self)
 {
     if (rb_block_given_p())
     {
-        // Run the block in self's context
-        (void) rb_obj_instance_eval(0, NULL, self);
+        if (rb_proc_arity(rb_block_proc()) == 1)
```

Comment:
I think following code is more better.

```c
    if (rb_block_given_p())
    {
        if (rb_proc_arity(rb_block_proc()) == 0)
        {
            // Run the block in self's context
            (void) rb_obj_instance_eval(0, NULL, self);
        }
        else
        {
            rb_yield(self);
        }
    }
```

Because `arity` might not == 1 with some case.
```
block = proc { |*a| }
p block.arity #=> -1
```

Reply:
Certainly that looks better